cmd/snap-fde-keymgr/main.go

```diff
@@ -76,8 +79,24 @@ func (c *cmdAddRecoveryKey) Execute(args []string) error {
 	// 3. add the key
 	// 4. if adding failed with keyslot already in used and the file was
 	// present assume it's correct
-	if err := keymgrAddRecoveryKeyToLUKSDevice(recoveryKey, c.Device); err != nil {
-		return fmt.Errorf("cannot add recovery key to LUKS device: %v", err)
+	if len(c.Authorization) != len(c.Device) {
+		return fmt.Errorf("cannot add recovery key: mismatch in the number of devices and authorizations")
+	}
+	for i, dev := range c.Device {
+		authz := c.Authorization[i]
+		if authz == "keyring" {
```

Comment:
(super nitpick) a `switch ` might look slightly nicer here (less busy to my eyes) instead of `if() {} else if()` (but maybe/probably just me)
```go
                switch {
		case if authz == "keyring":
			if err := keymgrAddRecoveryKeyToLUKSDevice(recoveryKey, dev); err != nil {
				return fmt.Errorf("cannot add recovery key to LUKS device: %v", err)
			}
		case strings.HasPrefix(authz, "file:"):
			authzKey, err := ioutil.ReadFile(authz[len("file:"):])
			if err != nil {
				return fmt.Errorf("cannot load authorization key: %v", err)
			}
			if err := keymgrAddRecoveryKeyToLUKSDeviceUsingKey(recoveryKey, authzKey, dev); err != nil {
				return fmt.Errorf("cannot add recovery key to LUKS device using authorization key: %v", err)
			}
		}
```